op-interop-filter/filter/frontend.go

```diff
@@ -0,0 +1,51 @@
+package filter
+
+import (
+	"context"
+	"errors"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rpc"
+
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/types"
+)
+
+// QueryFrontend handles supervisor query RPC methods
+type QueryFrontend struct {
+	backend *Backend
+}
+
+// CheckAccessList validates interop executing messages
+func (f *QueryFrontend) CheckAccessList(ctx context.Context, inboxEntries []common.Hash,
+	minSafety types.SafetyLevel, executingDescriptor types.ExecutingDescriptor) error {
+
+	err := f.backend.CheckAccessList(ctx, inboxEntries, minSafety, executingDescriptor)
+	if err != nil {
+		// Map errors to appropriate RPC error codes
+		code := types.GetErrorCode(err)
```

Comment:
(nit) It would be nice to gradually remove the op-supervisor dependency, because we will not use the supervisor. Can be done at the follow up PR.